modules/nw-configure-ingress-access-logging.adoc

```diff
@@ -0,0 +1,115 @@
+// Module included in the following assemblies:
+//
+// * ingress/configure-ingress-operator.adoc
+
+[id="nw-configure-ingress-access-logging_{context}"]
+= Configure Ingress access logging
+
+You can configure the Ingress Controller to enable access logs for the Ingress Controller. If you have clusters that do not receive much traffic, then you can log to a sidecar. If you have high traffic clusters, to avoid exceeding the capacity of the logging stack or integrate with a logging infrastructure outside of {product-title}, you can forward logs to a custom syslog endpoint. You can also specify the format for access logs.
+
+.Prerequisites
+
+* Log in as a user with `cluster-admin` privileges.
+
+.Procedure
+
+Configure Ingress access logging to a sidecar.
+
+* To configure Ingress access logging, you must specify a destination using `spec.logging.access.destination`. To specify logging to a sidecar container, you must specify `container` `spec.logging.access.destination.type`. The following example is the definition of an Ingress Controller that logs to a `container` destination:
++
+[source, yaml]
+----
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+metadata:
+  name: default
+  namespace: openshift-ingress-operator
+spec:
+  replicas: 2
+  endpointPublishingStrategy:
+    type: NodePortService <1>
+  logging:
+    access:
+      destination:
+        type: container
+----
+<1> `NodePortService` is not required to configure Ingress access logging to a sidecar. Ingress logging is compatible with any `endpointPublishingStrategy`.
++
+* When you create a sidecar container for logging, it creates a container named `logs` inside the router namespace. You can view the logs using the following command:
++
+----
+$ oc -n openshift-ingress logs  deployment.apps/router-internalapps -c logs
+----
++
+.Sample output:
+----
+2020-04-24T04:27:51.158772+00:00 router-internalapps-6bd9c8f99c-m8bhs router-internalapps-6bd9c8f99c-m8bhs haproxy[30]: Proxy public started.
+2020-04-24T04:27:51.158795+00:00 router-internalapps-6bd9c8f99c-m8bhs router-internalapps-6bd9c8f99c-m8bhs haproxy[30]: Proxy public_ssl started.
+2020-04-24T04:27:51.158800+00:00 router-internalapps-6bd9c8f99c-m8bhs router-internalapps-6bd9c8f99c-m8bhs haproxy[30]: Proxy be_sni started.
+2020-04-24T04:27:51.158814+00:00 router-internalapps-6bd9c8f99c-m8bhs router-internalapps-6bd9c8f99c-m8bhs haproxy[30]: Proxy fe_sni started.
```

Comment:
These lines are not actually access logs, but rather logging related to HAProxy's initialization.  Something like this would be a better example:
```
2020-05-11T19:11:50.135710+00:00 router-default-57dfc6cd95-bpmk6 router-default-57dfc6cd95-bpmk6 haproxy[108]: 174.19.21.82:39654 [11/May/2020:19:11:50.133] public be_http:hello-openshift:hello-openshift/pod:hello-openshift:hello-openshift:10.128.2.12:8080 0/0/1/0/1 200 142 - - --NI 1/1/0/0/0 0/0 "GET / HTTP/1.1"
```
My example above is for an Ingress Controller named "default", so the corresponding command would be `oc -n openshift-ingress logs  deployment.apps/router-default -c logs`.